### ISSUE TITLE
Add common missing ops.

### ISSUE THREAD

**Issue by nsthorat** [PAIR-code/deeplearnjs#223]
Flatten (covered by reshape, but would be a nice sugar method)
EmbeddingLookup
BatchNormalization2D
Conv1D, Conv3D
MaxPooling1D, MaxPooling2D (MaxPooling3D is already available)
Permute
Repeat
LeakyRELU, PRelu, ELU
Dropout (we could punt on this for now)

**Comment by dnnsthnnr** [comment]
I would also add axis aligned Softmax in multidimensional Arrays to the list.

**Comment by nsthorat** [comment]
We just committed the axis-aligned softmax:
https://github.com/PAIR-code/deeplearnjs/blob/master/src/math/math.ts#L782

**Comment by tycho01** [comment]
I'd found some node types listed on this [Neural Network Zoo](http://www.asimovinstitute.org/neural-network-zoo/) blog post of which I was wondering whether they're already covered among the current [graph methods](https://deeplearnjs.org/docs/api/classes/graph.html) as well.
I'm not sure all will correspond well since they're different layers of abstractions, but here's the impression I got comparing the two:
- [x] Input cell
- [ ] Back-fed input cell, listed as used as input of Hopfield Networks, (Restricted) Boltzmann Machine, Deep Belief Networks.
- [ ] Noisy input cell: listed as used as input of Denoising Auto-Encoders.
- [x] Hidden cell = Fully connected layer
- [ ] Probabilistic hidden cell: used in Markov Chains, (Restricted) Boltzmann Machine, Deep Belief Networks, Deep Convolutional Inverse Graphics Networks.
- [ ] Spiking hidden cell: used as the hidden cells of Liquid State Machines.
- [x] Output cell
- [ ] Match input-output cell: used in Auto Encoders, Deep Belief Networks, GANs.
- [x] Recurrent cell = ReLU
- [ ] Memory cell: used in LSTMs, Neural Turing Machines
- [ ] Different memory cell: used in Gated Recurrent Units
- [ ] Kernel cell: used in conv nets.
- [x] Convolution

I'm probably wrong about half of these as I know [conv nets](https://deeplearnjs.org/demos/model-builder/) and [GANs](https://reiinakano.github.io/gan-playground) were already being made with this, but maybe some others are still missing, if maybe not as common.

**Comment by dsmilkov** [comment]
Hi,

With the [release of 0.5.0](https://github.com/PAIR-code/deeplearnjs/releases/tag/v0.5.0) we are deprecating the graph API, in favor of the eager API. Feel free to reopen another issue for any op that is missing in the new API. Doc reference [here](https://deeplearnjs.org/docs/api/index.html).
